File: tut/src/param/param_test.py
#!/usr/bin/python3

import rospy as ros


def main():
    ros.init_node('hello_py')
    ros.sleep(1)

    ros.set_param("/foo", "Hello")
    ros.set_param("/bar", 1)

    if (ros.has_param("/foo") and ros.has_param("/bar")):
        foo = ros.get_param("/foo")
        bar = ros.get_param("/bar")
        ros.loginfo(f"{foo} {bar}")
    else:
        ros.logerr("Failed to get the parameters")
        exit(1)


if (__name__ == "__main__"):
    main()

File: tut/src/param/param_test.cpp
#include <ros/ros.h>
#include <string>


int main(int argc, char** argv)
{
    ros::init(argc, argv, "param_test");
    ros::NodeHandle nh;
    ros::Duration(1).sleep();

    nh.setParam("/foo", "Hello");
    nh.setParam("/bar", 1);

    std::string foo = "";
    int bar = 0;

    if (nh.getParam("/foo", foo) && nh.getParam("/bar", bar)) {
        ROS_INFO("%s %d", foo.c_str(), bar);
    }
    else{
        ROS_ERROR("Failed to get the parameters");
        return 1;
    }

    return 0;
}
